stock_delivery_notification/models/stock_picking.py

```diff
@@ -35,13 +35,13 @@ def action_send_delivery_order(self):
         self.ensure_one()
         if self.picking_type_id.code == "outgoing" and self.group_id:
```

Comment:
```
        if self.picking_type_id.code == "outgoing" and self.sale_id and self.group_id:
```

Haven't tested, but I'd personally prefer this over your suggestion as it'd make the context clearer.

Reply:
I think we can further simplify the code using self.sale_id, actually (our code is redundant with Odoo's standard logic).

See how sale_id is computed here: https://github.com/odoo/odoo/blob/6fa705399bc05d4b0dce5e7181c40a4a5ab8b1c3/addons/sale_stock/models/stock.py#L51-L55